apps/web/src/hooks/useRegisterNameCallback.ts

```diff
@@ -114,16 +157,35 @@ export function useRegisterNameCallback(
       ],
     });
 
+    let coinTypesForRequest: readonly bigint[] = [];
+    let signatureExpiryForRequest: bigint = 0n;
+    let signatureForRequest: `0x${string}` = '0x';
+
+    if (!paymasterServiceEnabled && reverseRecord) {
+      const payload =
+        reverseSigPayload ??
+        (await prepareReverseRecordSignature().catch((e) => {
+          logError(e, 'Reverse record signature step failed');
+          return null;
+        }));
+      if (!payload) return;
+      coinTypesForRequest = payload.coinTypes;
+      signatureExpiryForRequest = payload.signatureExpiry;
+      signatureForRequest = payload.signature;
```

Comment:
I think we can remove the `reverseSigPayload` state and do this:

```suggestion
      try {
        const payload = await prepareReverseRecordSignature().
        coinTypesForRequest = payload?.coinTypes ?? [];
        signatureExpiryForRequest = payload?.signatureExpiry ?? 0n;
        signatureForRequest = payload?.signature ?? '0x';
      } catch (e) {
        logError(e, 'Reverse record signature step failed');
      }
```